### ISSUE TITLE
linter: CLI -D does not override severity for jsPlugins rules

### ISSUE THREAD

**Issue by yamachan0625** [oxc-project/oxc#22730]
### What version of Oxlint are you using?

1.64.0

### What command did you run?

```sh
oxlint -c .oxlintrc.json repro.ts --format json
oxlint -c .oxlintrc.json -D import-js/order repro.ts --format json
```

For comparison, the same CLI override works for a native Oxlint rule:

```sh
oxlint -c .oxlintrc.json -D no-unused-vars repro.ts --format json
```

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```jsonc
{
  "$schema": "./node_modules/oxlint/configuration_schema.json",
  "jsPlugins": [
    {
      "name": "import-js",
      "specifier": "eslint-plugin-import"
    }
  ],
  "rules": {
    "no-unused-vars": "warn",
    "import-js/order": [
      "warn",
      {
        "groups": ["builtin", "external", "parent", "sibling", "index", "object", "type"],
        "newlines-between": "never",
        "alphabetize": {
          "order": "asc",
          "caseInsensitive": true
        }
      }
    ]
  }
}
```

### What happened?

`-D import-js/order` does not change the `import-js/order` diagnostic from warning to error.

Minimal repro:

```ts
import { join } from "node:path";
import { existsSync } from "node:fs";

const unused = 1;

export const repro = [existsSync("."), join("a", "b")];
```

Without `-D import-js/order`, `import-js/order` is reported as a warning, as configured:

```json
{
  "message": "`node:fs` import should occur before import of `node:path`",
  "code": "import-js(order)",
  "severity": "warning"
}
```

With `-D import-js/order`, the diagnostic is still reported as a warning:

```json
{
  "message": "`node:fs` import should occur before import of `node:path`",
  "code": "import-js(order)",
  "severity": "warning"
}
```

For comparison, `-D no-unused-vars` correctly changes a native Oxlint rule from warning to error:

```json
{
  "message": "Variable 'unused' is declared but never used. Unused variables should start with a '_'.",
  "code": "eslint(no-unused-vars)",
  "severity": "error"
}
```

### Expected behavior

`-D import-js/order` should override the severity of the `jsPlugins` rule, the same way it does for built-in/native Oxlint rules.

If CLI severity overrides are intentionally unsupported for `jsPlugins` rules, Oxlint should document that limitation or report that the rule id cannot be resolved by CLI flags.

### Why this matters

This makes it hard to run a JS plugin rule as warning during normal linting, but as error during a formatting/fix script.

In our case, we want `import-js/order` to be warning in regular lint, but error during `--fix`, so a retry loop can continue while auto-fixable import-order diagnostics remain.

This is related to the workaround described in #22725 / #16118, but the issue here is separate: CLI severity overrides do not appear to apply to `jsPlugins` rule ids.
